website/members/models/member.py

```diff
@@ -117,6 +117,17 @@ def latest_membership(self):
             return None
         return self.membership_set.latest("since")
 
+    @property
+    def oldest_membership_of_current_type(self):
+        """Get the first membership of the most recent membership's type."""
+        if not self.membership_set.exists():
+            return None
+        return (
+            self.membership_set.filter(type=self.latest_membership.type)
```

Comment:
What happens when a user is a member from 2014-09-01 until 2015-08-31, then a benefactor from 2015-09-01 until 2018-08-31, and then a member again starting 2018-09-01?

I don't think it should show `Member (since 1 Sept 2014)` in that case. Since that is not true. And I feel that this code would do that.

Reply:
It would display `Member (since 1 Sept 2014)` indeed.
I will look into placing this in the membership, and I'll probably change it to be consecutive memberships of the same type.

Reply:
https://github.com/svthalia/concrexit/blob/master/website/activemembers/models.py#L234